Using a reserve word "char" in JS

Using a reserved word in Javascript: **char** in the following function.

```javascript
  replaceFullWidthChars = function(str) {
    var char, chars, fullWidth, halfWidth, idx, value, _i, _len;
    if (str == null) {
      str = '';
    }
    fullWidth = '\uff10\uff11\uff12\uff13\uff14\uff15\uff16\uff17\uff18\uff19';
    halfWidth = '0123456789';
    value = '';
    chars = str.split('');
    for (_i = 0, _len = chars.length; _i < _len; _i++) {
      char = chars[_i];
      idx = fullWidth.indexOf(char);
      if (idx > -1) {
        char = halfWidth[idx];
      }
      value += char;
    }
    return value;
  };
```

This gives me the following js error when compressing with YUI Compressor:
[ERROR] in -
  175:13:missing variable name
[ERROR] in -
  184:11:identifier is a reserved word
[ERROR] in -
  185:35:identifier is a reserved word
[ERROR] in -
  187:13:identifier is a reserved word
[ERROR] in -
  189:20:identifier is a reserved word
[ERROR] in -
  1:0:Compilation produced 5 syntax errors.

--- a/lib/jquery.payment.js
+++ b/lib/jquery.payment.js
@@ -172,7 +172,7 @@
   };
 
   replaceFullWidthChars = function(str) {
-    var char, chars, fullWidth, halfWidth, idx, value, _i, _len;
+    var chars, chr, fullWidth, halfWidth, idx, value, _i, _len;
     if (str == null) {
       str = '';
     }
@@ -181,12 +181,12 @@
     value = '';
     chars = str.split('');
     for (_i = 0, _len = chars.length; _i < _len; _i++) {
-      char = chars[_i];
-      idx = fullWidth.indexOf(char);
+      chr = chars[_i];
+      idx = fullWidth.indexOf(chr);
       if (idx > -1) {
-        char = halfWidth[idx];
+        chr = halfWidth[idx];
       }
-      value += char;
+      value += chr;
     }
     return value;
   };

--- a/src/jquery.payment.coffee
+++ b/src/jquery.payment.coffee
@@ -159,10 +159,11 @@ replaceFullWidthChars = (str = '') ->
   value = ''
   chars = str.split('')
 
-  for char in chars
-    idx = fullWidth.indexOf(char)
-    char = halfWidth[idx] if idx > -1
-    value += char
+  # Avoid using reserved word `char`
+  for chr in chars
+    idx = fullWidth.indexOf(chr)
+    chr = halfWidth[idx] if idx > -1
+    value += chr
 
   value
 

